kubernetes/kubeclient.go

```diff
@@ -96,10 +99,33 @@ func VampConfigValidateAndSetupDefaults(config *models.VampConfig) (*models.Vamp
 		config.IstioInstallerImage = DefaultVampConfig.IstioInstallerImage
 		fmt.Printf("Istio Installer Image set to default value: %v\n", config.IstioInstallerImage)
 	}
-	if config.MaxReplicas == 0 {
-		config.MaxReplicas = 6
+	if config.MinReplicas != nil && *config.MinReplicas <= 0 {
+		config.MinReplicas = DefaultVampConfig.MinReplicas
+		fmt.Printf("MinReplicas set to default value %v\n", func() interface{} {
+			if config.MinReplicas != nil {
+				return *config.MinReplicas
+			}
+			return "K8s default value"
+		}())
+	}
+	if config.MaxReplicas <= 0 {
+		config.MaxReplicas = DefaultVampConfig.MaxReplicas
+		fmt.Printf("MaxReplicas set to default %v\n", config.MaxReplicas)
+	}
+	// MaxReplicas cannot be smaller than MinReplicas
+	if config.MinReplicas != nil && config.MaxReplicas < *config.MinReplicas {
+		config.MaxReplicas = *config.MinReplicas
 		fmt.Printf("MaxReplicas set to %v\n", config.MaxReplicas)
 	}
+	if config.TargetCPUUtilizationPercentage != nil && *config.TargetCPUUtilizationPercentage <= 0 {
```

Comment:
Also, can you check if it is not more than 100?

Reply:
The thing is it looks like it can be greater than 100. At least I saw such values (current cpu utilization) in output of `kubectl get hpa --all-namespaces`.

Reply:
Let's merge this then create another ticket to move hpa to metric scaler since TargetCPUUtilizationPercentage is removed in the latest one. TargetCPUUtilizationPercentage is a percentage of the resource limits so it should never pass  100. The current CPU utilization is can be bigger than 100. They removed the official documentation.